stdlib/tests/collections/smt.rs

```diff
@@ -0,0 +1,124 @@
+    IntoBytes, MerkleStore, NodeIndex, Rpo256, StarkField, Word, WORD_SIZE, ZERO,
+};
+
+#[test]
+fn foo() {
+    let seed = [0xff; 32];
+    let a: Word = test_utils::rand::prng_array(seed);
+    let b: Word = test_utils::rand::prng_array(seed);
+    assert_eq!(a, b);
+}
+
+proptest! {
+    #[test]
+    fn arbitrary_smtget_opens_correctly(
+        ref seed in any::<[u8; 32]>()
+    ) {
+        // setup the test
+        let empty = EmptySubtreeRoots::empty_hashes(64);
+        let initial_root = Word::from(empty[0]);
+        let mut seed = *seed;
+        let key = seeded_word(&mut seed);
+        let value = seeded_word(&mut seed);
+
+        // check empty tree
+        let store = MerkleStore::new();
+        let advice_map = [];
+        let expected = [ZERO; WORD_SIZE];
+        assert_case_arbitrary_smtget_opens_correctly(
+            key,
+            expected,
+            initial_root,
+            store,
+            &advice_map
+        );
+
+        // check included leaves for all tiers
+        for depth in [16, 32, 48, 64] {
+            // compute the index and node value
+            let index = key[3].as_int();
+            let index = index >> (64 - depth);
+            let index = NodeIndex::new(depth as u8, index).unwrap();
+            let remaining = get_remaining_key(key, depth as u8);
+            let node =
+                Rpo256::merge_in_domain(&[remaining.into(), value.into()], Felt::new(depth)).into();
+
+            // setup the store and run the test
+            let mut store = MerkleStore::new();
+            let root = store.set_node(initial_root, index, node).unwrap().root;
+            let advice_map = [(node.into_bytes(), value.to_vec())];
+            assert_case_arbitrary_smtget_opens_correctly(
+                key,
+                value,
+                root,
+                store,
+                &advice_map
+            );
+        }
+
+        // check non-included leaf with same path sibling, except for last bit
+        {
+            // compute a sibling index and node value
+            let mut sibling = key[3].as_int();
+            sibling ^= 1;
+            let remaining = get_remaining_key(key, 64);
+            let sibling = NodeIndex::new(64, sibling).unwrap();
+            let node = Rpo256::merge_in_domain(&[remaining.into(), value.into()], Felt::new(64)).into();
+
+            // setup the store and run the test; expect ZERO as the leaf isn't included
+            let mut store = MerkleStore::new();
+            let root = store.set_node(initial_root, sibling, node).unwrap().root;
+            let advice_map = [(node.into_bytes(), value.to_vec())];
+            let expected = [ZERO; WORD_SIZE];
+            assert_case_arbitrary_smtget_opens_correctly(
+                key,
+                expected,
+                root,
+                store,
+                &advice_map
+            );
+        }
```

Comment:
I'd probably move this out into its own test (doesn't have to be proptest). This test would work as follows:

1. Similar test to what we have here - i.e., a tree has one node and we do a lookup for a key with a shared prefix. I think one thing I'd change is to test for prefixes of different shared length (i.e., 16, 32, 48, 64).
2. Initialize the tree which has two nodes which share the same prefix (i.e., 16, 32, 48, or 64 bits), and then do a lookup for a key which shares the same prefix.